src/toolchain-conventions.adoc

```diff
@@ -1,3 +1,17 @@
+== Target Triples
+
+The toolchains have agreed on the following architecture names to be used for
+RISC-V in the first segment of target triples:
+
+- `riscv32` RV32, little-endian
+- `riscv32be` RV32, big-endian
+- `riscv64` RV64, little-endian
+- `riscv64be` RV64, big-endian
+
```

Comment:
config.sub in the autoconf repo also knows `riscv` (see https://github.com/autotools-mirror/autoconf/blob/master/build-aux/config.sub#L1239).
But I don't know if that is supported by tools.

Reply:
Clang doesn't seem to know about it:
```
$ clang --target=riscv -march=rv32gc -print-effective-triple
riscv
$ clang --target=riscv64 -march=rv32gc -print-effective-triple
riscv32--
```

(What this shows is that clang is canonicalising the triplet when the march/triplet arch disagree with the 63/32 suffix, but isn't touching the triplet arch at all when the arch is just riscv, because it is not recognised). The llvm Triple(t) docs point to `config.guess`, which doesn't seem to mention just `riscv`. The code to make the triplet switch based on `-march` is here: https://github.com/llvm/llvm-project/blob/2a702cdc38f96989cb41c0b863d977f0d8f4f325/clang/lib/Driver/Driver.cpp#L845